js/shared.js

```diff
@@ -2003,4 +2003,41 @@ export const getSelectedLanguage = () => {
     }
 
     return selectedLanguage;
-}
\ No newline at end of file
+}
+
+/**
+ * Get the custom settings for ConnectApp. Initial use: Quest versioning. See loadQuestConfig().
+ * @param {Array<string>} paramsToFetchArray - Optional array of parameters to fetch. E.g. ['param1', 'param2', 'param3']. Omit to return entire appSettings object.
+ * @returns {Object} - App settings object.
+ */
+export const getAppSettings = async (paramsToFetchArray) => {
+    const queryParams = paramsToFetchArray && paramsToFetchArray.length > 0
+        ? `&selectedParamsArray=${encodeURIComponent(paramsToFetchArray.map(param => param.trim()).join(','))}`
+        : '';
+
+    const url = `${api}?api=getAppSettings${queryParams ? `${queryParams}` : ''}`;
+
+    try {
+        const idToken = await getIdToken();
+        const response = await fetch(url, {
+            method: "GET",
+            headers: {
+                Authorization: "Bearer " + idToken,
+            },
+        });
+    
+        if (!response.ok) {
+            throw new Error(`Server responded with: ${response.status}`);
+        }
+    
+        const jsonResponse = await response.json();
+        
+        if (jsonResponse.code === 200) {
```

Comment:
Are the checks on lines 2029 and 2035 redundant? It doesn't look like the endpoint you created would return a code between 201-209.

Reply:
Good point -- we really only need to check the `.code === 200`. Updated.

Reply:
Note: I also just updated the Error syntax